proxy/src/connection.rs

```diff
@@ -241,6 +241,9 @@ impl Connection {
 
 impl io::Read for Connection {
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+        // TODO: Eliminate the duplication between this and
```

Comment:
Am I correct that `Connection` could eventually just be something like
```rust
pub struct Connection {
    io: Prefixed<BoxedIo>,
    tls_status: ...
}
```

Reply:
I think that's right. Unfortunately, abstracting this so that it can be used in both places is not as simple as it seems at first, so I'm deferring that.

Reply:
Okay, I don't think it's necessary to make that change now, then; I was just curious if it was something we would be able to do in the future.